client/my-sites/stats/post-detail-highlights-section/index.tsx

```diff
@@ -27,16 +27,34 @@ type Post = {
 
 const POST_STATS_CARD_TITLE_LIMIT = 48;
 
-// Use ellipsis when characters count over the limit
-const textTruncator = ( text: string, limit = 48 ) => {
-	if ( ! text ) {
-		return '';
+function truncateWithLimit( text: string, limit: number ): string {
+	// Determine if any processing is needed.
+	const trimmedText = text.trim();
+	if ( trimmedText.length <= limit ) {
+		return trimmedText;
 	}
 
-	const truncatedText = text.substring( 0, limit );
+	// Find the last whitespace character within the limit.
+	const truncatedText = trimmedText.substring( 0, limit );
+	const lastWhitespaceIndex = truncatedText.lastIndexOf( ' ' );
 
-	return `${ truncatedText }${ text.length > limit ? '...' : '' } `;
-};
+	// If there's no whitespace within the limit, truncate at the limit.
+	if ( lastWhitespaceIndex === -1 ) {
+		return truncatedText + '...';
+	}
+
+	// Truncate at the last whitespace character.
+	return trimmedText.substring( 0, lastWhitespaceIndex ) + '...';
+}
+
+function getProcessedTitle( post: Post ): string {
+	const title = post?.title || '';
+	if ( ! title || title.length === 0 ) {
```

Comment:
```suggestion
	if ( ! title ) {
```
I think the `! title` check can handle the case of `title.length === 0`, isn't it?

Reply:
Aha, you are correct. Thanks for the catch, Dognose!